src/eventDispatcher.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+function EventDispatcher() {
+    this.eventDispatcherMap = {};
+};
+
+
+EventDispatcher.prototype.bind = function(eventName, handler) {
+    if (typeof handler === 'function') {
+        if (this.eventDispatcherMap[eventName]) {
+            this.eventDispatcherMap[eventName].push(handler);
+        } else {
+            this.eventDispatcherMap[eventName] = [handler];
+        }
+    }
+}
+
+EventDispatcher.prototype.trigger = function(eventName) {
+    if (this.eventDispatcherMap[eventName]) {
+        for (var i = 0; i < this.eventDispatcherMap[eventName].length; i++) {
+            try {
+                this.eventDispatcherMap[eventName][i]();
+            } catch (e) {
+                this.eventDispatcherMap[eventName].splice(i, 1);
```

Comment:
Great work, I'm really proud :)

One important thing, we use artificial examples here. In reality, a function may or may not throw an exception, depends from various combinations. At one time it may throw an error, where in another it may not. Having that in mind, you should not exclude them from array here. You should just catch an error and ignore it.